taxonomy-manager-engine/src/main/java/com/digirati/taxman/analysis/nlp/corenlp/CoreNlpWordTokenizer.java

```diff
@@ -49,9 +53,13 @@ public static WordTokenizer create(String language) {
 
     @Override
     public List<WordToken> tokenize(String input) {
+        // Strip HTML tags and replace them with equivalent amount of '_' chars
+        HtmlTagReplacer.matcher(input).replaceAll(
+                matchResult -> StringUtils.repeat('_', matchResult.end() - matchResult.start()));
+
         // Some input strings like to delimit text with a forward slash. This causes problems for Stanford CoreNLP,
         // which expects words to be delimited by whitespace or regular punctuation.
-        final String normalizedInput = input.replaceAll("/", " / ");
+        final String normalizedInput = input.replaceAll("/", " ");
```

Comment:
This looks like it can potentially change matching semantics. Will leave that for regression testing to identify.

Reply:
It was specifically done because the process looks like this:

CRUPlatform has an `Article (Title, Subtitle, Teaser, Body, CruViewComment)`

CRUPlatform creates a string like `Title + " " + Subtitle + " " + ... + CruViewComment`

CRUPlatform notes down the offset at which each "segment" starts in the constructed string

CRUPlatform sends string to TaxMan

TaxMan provides position(s) in the submitted string where the concept was found

CRUPlatform uses the previously saved offsets to translate the TaxMan position into "which field at which position" information

So the TaxMan _cannot_ modify the length of the text before it is tokenized.

Reply:
Hm, this is something we should look at supporting in another way in future. Happy with this for now.